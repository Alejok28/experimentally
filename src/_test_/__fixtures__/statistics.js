export const statistics = {
  "viewCount": "28640082",
  "likeCount": "556518",
  "dislikeCount": "6249",
  "favoriteCount": "0",
  "commentCount": "12663"
}