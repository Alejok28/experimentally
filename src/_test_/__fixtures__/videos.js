export const videos = [
  {
    "kind": "youtube#searchResult",
    "etag": "\"nlUZBA6NbTS7q9G8D1GljyfTIWI/n8La5FHPnJZ9SjefY7NHLK55KlY\"",
    "id": {
      "kind": "youtube#video",
      "videoId": "z2VZ_OGz0RU"
    },
    "snippet": {
      "publishedAt": "2017-03-22T10:01:08.000Z",
      "channelId": "UCftwRNsjfRo08xYE31tkiyw",
      "title": "Ryan Reynolds &amp; Jake Gyllenhaal Answer the Web&#39;s Most Searched Questions | WIRED",
      "description": "'Life' stars Ryan Reynolds and Jake Gyllenhaal waste some time with WIRED and answer the Internet's most searched questions about themselves. Does Ryan ...",
      "thumbnails": {
        "default": {
          "url": "https://i.ytimg.com/vi/z2VZ_OGz0RU/default.jpg",
          "width": 120,
          "height": 90
        },
        "medium": {
          "url": "https://i.ytimg.com/vi/z2VZ_OGz0RU/mqdefault.jpg",
          "width": 320,
          "height": 180
        },
        "high": {
          "url": "https://i.ytimg.com/vi/z2VZ_OGz0RU/hqdefault.jpg",
          "width": 480,
          "height": 360
        }
      },
      "channelTitle": "WIRED",
      "liveBroadcastContent": "none"
    }
  },
  {
    "kind": "youtube#searchResult",
    "etag": "\"nlUZBA6NbTS7q9G8D1GljyfTIWI/KPeWEEZzKXm-pfv-1ywpZ2WOtPE\"",
    "id": {
      "kind": "youtube#video",
      "videoId": "F9CrRG6j2SM"
    },
    "snippet": {
      "publishedAt": "2019-08-05T09:00:02.000Z",
      "channelId": "UCfkXDY7vwkcJ8ddFGz8KusA",
      "title": "[M/V] SEVENTEEN(세븐틴) - HIT",
      "description": "13명의 뚜렷한 음악 색을 바탕으로 세븐틴 표 노래와 무대를 전 세계에 알리며 독보적인 독창성을 보여주고 있는 자체 제작 아이돌 세븐틴이 또...",
      "thumbnails": {
        "default": {
          "url": "https://i.ytimg.com/vi/F9CrRG6j2SM/default.jpg",
          "width": 120,
          "height": 90
        },
        "medium": {
          "url": "https://i.ytimg.com/vi/F9CrRG6j2SM/mqdefault.jpg",
          "width": 320,
          "height": 180
        },
        "high": {
          "url": "https://i.ytimg.com/vi/F9CrRG6j2SM/hqdefault.jpg",
          "width": 480,
          "height": 360
        }
      },
      "channelTitle": "SEVENTEEN",
      "liveBroadcastContent": "none"
    }
  },
  {
    "kind": "youtube#searchResult",
    "etag": "\"nlUZBA6NbTS7q9G8D1GljyfTIWI/id3uz64VI_Frb6lxlzAYDG8yCuw\"",
    "id": {
      "kind": "youtube#video",
      "videoId": "A2FsgKoGD04"
    },
    "snippet": {
      "publishedAt": "2017-05-25T21:58:58.000Z",
      "channelId": "UCRijo3ddMTht_IHyNSNXpNQ",
      "title": "Real Life Trick Shots | Dude Perfect",
      "description": "Trick shots should be an everyday thing! Thanks to Kingsford for sponsoring this video! ▻ Click HERE to subscribe to Dude Perfect! http://bit.ly/SubDudePerfect ...",
      "thumbnails": {
        "default": {
          "url": "https://i.ytimg.com/vi/A2FsgKoGD04/default.jpg",
          "width": 120,
          "height": 90
        },
        "medium": {
          "url": "https://i.ytimg.com/vi/A2FsgKoGD04/mqdefault.jpg",
          "width": 320,
          "height": 180
        },
        "high": {
          "url": "https://i.ytimg.com/vi/A2FsgKoGD04/hqdefault.jpg",
          "width": 480,
          "height": 360
        }
      },
      "channelTitle": "Dude Perfect",
      "liveBroadcastContent": "none"
    }
  },
  {
    "kind": "youtube#searchResult",
    "etag": "\"nlUZBA6NbTS7q9G8D1GljyfTIWI/mGGEbULKJWR5g9X1nJ2MGKv0MKo\"",
    "id": {
      "kind": "youtube#video",
      "videoId": "t_qn-f7XfJo"
    },
    "snippet": {
      "publishedAt": "2019-04-26T20:58:47.000Z",
      "channelId": "UCte53PewI8_jYpBYn071MqQ",
      "title": "Sech - Otro Trago ft. Darell (Video Oficial)",
      "description": "Escúchalo en Spotify: https://open.spotify.com/track/1Ej96GIBCTvgH7tNX1r3qr?si=xhlKQ1ZGR6iT5xnErFxojA Producido por Cinema Entertainment Sigue a ...",
      "thumbnails": {
        "default": {
          "url": "https://i.ytimg.com/vi/t_qn-f7XfJo/default.jpg",
          "width": 120,
          "height": 90
        },
        "medium": {
          "url": "https://i.ytimg.com/vi/t_qn-f7XfJo/mqdefault.jpg",
          "width": 320,
          "height": 180
        },
        "high": {
          "url": "https://i.ytimg.com/vi/t_qn-f7XfJo/hqdefault.jpg",
          "width": 480,
          "height": 360
        }
      },
      "channelTitle": "Sech",
      "liveBroadcastContent": "none"
    }
  },
  {
    "kind": "youtube#searchResult",
    "etag": "\"nlUZBA6NbTS7q9G8D1GljyfTIWI/1szGxIC2JpHOZayLDDyWY1wRMRg\"",
    "id": {
      "kind": "youtube#video",
      "videoId": "VRJmcxCrAOA"
    },
    "snippet": {
      "publishedAt": "2018-01-22T22:55:24.000Z",
      "channelId": "UCRijo3ddMTht_IHyNSNXpNQ",
      "title": "Real Life Trick Shots 2 | Dude Perfect",
      "description": "Trick shots should be an everyday thing! Thanks to Kay Jewelers for sponsoring this video! Win Valentine's Day by going to http://dudeperfect.com/KayJewelers ...",
      "thumbnails": {
        "default": {
          "url": "https://i.ytimg.com/vi/VRJmcxCrAOA/default.jpg",
          "width": 120,
          "height": 90
        },
        "medium": {
          "url": "https://i.ytimg.com/vi/VRJmcxCrAOA/mqdefault.jpg",
          "width": 320,
          "height": 180
        },
        "high": {
          "url": "https://i.ytimg.com/vi/VRJmcxCrAOA/hqdefault.jpg",
          "width": 480,
          "height": 360
        }
      },
      "channelTitle": "Dude Perfect",
      "liveBroadcastContent": "none"
    }
  },
  {
    "kind": "youtube#searchResult",
    "etag": "\"nlUZBA6NbTS7q9G8D1GljyfTIWI/0WX426ECapOMYBfaUd1p8CnScQ8\"",
    "id": {
      "kind": "youtube#video",
      "videoId": "rtgY1q0J_TQ"
    },
    "snippet": {
      "publishedAt": "2014-09-22T23:08:57.000Z",
      "channelId": "UCUsN5ZwHx2kILm84-jPDeXw",
      "title": "Key &amp; Peele - Gay Wedding Advice",
      "description": "In preparation for their cousin's marriage to another man, a family asks a gay man questions about what the wedding ceremony will be like and for \"advice\" about ...",
      "thumbnails": {
        "default": {
          "url": "https://i.ytimg.com/vi/rtgY1q0J_TQ/default.jpg",
          "width": 120,
          "height": 90
        },
        "medium": {
          "url": "https://i.ytimg.com/vi/rtgY1q0J_TQ/mqdefault.jpg",
          "width": 320,
          "height": 180
        },
        "high": {
          "url": "https://i.ytimg.com/vi/rtgY1q0J_TQ/hqdefault.jpg",
          "width": 480,
          "height": 360
        }
      },
      "channelTitle": "Comedy Central",
      "liveBroadcastContent": "none"
    }
  }
]