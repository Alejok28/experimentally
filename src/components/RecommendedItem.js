import React from 'react';
import { makeStyles } from '@material-ui/styles';
import PropTypes from 'prop-types';

// Components
import Grid from '@material-ui/core/Grid';
import Tooltip from '@material-ui/core/Tooltip';

// Others
import { escapeHtml } from '../utils/functions';

const useStyles = makeStyles({
  grid: {
    margin: '10px'
  },
  img: {
    cursor: 'pointer'
  }
});

export default function RecommendedItem({ video, handleClick }) {
  const classes = useStyles();
  const { snippet } = video;

  return (
    <Grid className={classes.grid} item xs={12}>
      <Tooltip title={escapeHtml(snippet.title)}>
        <img
          className={classes.img}
          src={snippet.thumbnails.default.url}
          alt={snippet.title}
          onClick={() => handleClick(video)}
        />
      </Tooltip>
    </Grid>
  )
}


RecommendedItem.propTypes = {
  video: PropTypes.object.isRequired,
  handleClick: PropTypes.func
};
