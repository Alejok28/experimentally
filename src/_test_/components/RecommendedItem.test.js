import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import RecommendedItem from '../../components/RecommendedItem';
import { videos } from '../__fixtures__/videos';

afterEach(cleanup);

test('renders image corrrectly', () => {
  const { container } = render(<RecommendedItem video={videos[0]} />);
  const image = container.querySelector('img');
  expect(image).toHaveProperty('alt', videos[0].snippet.title);
  expect(image).toHaveProperty('src', videos[0].snippet.thumbnails.default.url);
  expect(container).toMatchSnapshot();
});

test('calls "onClick" prop on image click', () => {
  const handleClick = jest.fn();
  const { container } = render(
    <RecommendedItem video={videos[0]} handleClick={handleClick} />
  );
  const image = container.querySelector('img');
  fireEvent.click(image);
  expect(handleClick.mock.calls.length).toBe(1);
});
