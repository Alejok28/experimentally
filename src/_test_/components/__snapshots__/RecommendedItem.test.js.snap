// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders image corrrectly 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-grid-1 MuiGrid-item MuiGrid-grid-xs-12"
  >
    <img
      alt="Ryan Reynolds &amp; Jake Gyllenhaal Answer the Web&#39;s Most Searched Questions | WIRED"
      class="makeStyles-img-2"
      src="https://i.ytimg.com/vi/z2VZ_OGz0RU/default.jpg"
      title="Ryan Reynolds & Jake Gyllenhaal Answer the Web's Most Searched Questions | WIRED"
    />
  </div>
</div>
`;
